node/pkg/allocateip/allocateip_test.go

```diff
@@ -372,51 +420,88 @@ var _ = allocateIPDescribe("ensureHostTunnelAddress", []string{ipam.AttributeTyp
 	var be bapi.Client
 	BeforeEach(func() {
 		// Clear out datastore
-		be, err := backend.NewClient(*cfg)
+		var err error
+		be, err = backend.NewClient(*cfg)
 		Expect(err).ToNot(HaveOccurred())
 		err = be.Clean()
 		Expect(err).ToNot(HaveOccurred())
 
 		//create client.
 		c, _ = client.New(*cfg)
 
-		//create IPPool which has only two ips available.
-		_, err = c.IPPools().Create(ctx, makeIPv4Pool("pool1", "172.16.0.0/31", 31), options.SetOptions{})
+		pool1CIDR, _, pool2CIDR, _ := getEnsureCIDRAndExpectedAddrForIPVersion(4)
+
+		//create IPPool which has only one ip available.
+		_, err = c.IPPools().Create(ctx, makeIPPool("pool1-v4", pool1CIDR, 32, api.IPIPModeAlways, api.VXLANModeNever), options.SetOptions{})
 		Expect(err).NotTo(HaveOccurred())
 
-		//create second IPPool which has only one ips available.
-		_, err = c.IPPools().Create(ctx, makeIPv4Pool("pool2", "172.16.10.10/32", 32), options.SetOptions{})
+		// create IPPool which has only two ips available.
+		_, err = c.IPPools().Create(ctx, makeIPPool("pool2-v4", pool2CIDR, 31, api.IPIPModeAlways, api.VXLANModeNever), options.SetOptions{})
 		Expect(err).NotTo(HaveOccurred())
 
 		// Pre-allocate a WEP ip on 172.16.0.0. This will force tunnel address to use 172.16.0.1
 		handle := "myhandle"
-		wepIp := gnet.IP{172, 16, 0, 0}
 		wepAttr = map[string]string{
 			ipam.AttributeNode: "test.node",
 			ipam.AttributeType: ipam.AttributePod,
 		}
 		err = c.IPAM().AssignIP(ctx, ipam.AssignIPArgs{
-			IP:       net.IP{IP: wepIp},
+			IP:       net.IP{IP: gnet.ParseIP("172.16.0.0")},
 			Hostname: "test.node",
 			HandleID: &handle,
 			Attrs:    wepAttr,
 		})
 		Expect(err).NotTo(HaveOccurred())
+
+		//TODO: enable when libcalico-go validator no longer gives an error for VXLAN IPv6 IPPools
```

Comment:
Should probably include this change in this PR so we can uncommend this. WDYT?

Reply:
My intention was to change the validation last, as there are still quite a bit of changes needed for IPv6 VXLAN support, so we would keep them unexposed until full support is in. Do you think that's not a good idea?